src/utils/timeline.rkt

```diff
@@ -35,11 +35,14 @@
 
 (define (timeline-event! type)
   (when *timeline-active-key*
-    (hash-update! (car (unbox (*timeline*)))
-                  *timeline-active-key*
-                  (curry append *timeline-active-value*)
-                  '())
-    (set! *timeline-active-key* #f))
+    (if (pair? (unbox (*timeline*)))
+        (lambda ()
+          (hash-update! (car (unbox (*timeline*)))
+                        *timeline-active-key*
+                        (curry append *timeline-active-value*)
+                        '())
+          (set! *timeline-active-key* #f))
+        (error 'timeline "Cannot push to an empty timeline")))
```

Comment:
No no, don't do this here, the whole point is we want to do this in `timeline-push!` which is the actual thing that breaks the invariant. See the Slack thread for where to actually put this code.

Also you wrapped the "then" clause in a lambda, which means it won't actually be executed, which is bad, we'll drop timeline entries.

Reply:
Thank you. I completely read the Slack wrong.